protocol/x/clob/types/match_with_orders.go

```diff
@@ -76,5 +76,10 @@ func (match MatchWithOrders) Validate() error {
 	if makerOrder.MustGetOrder().TimeInForce == Order_TIME_IN_FORCE_IOC {
 		return errors.New("IOC order cannot be matched as a maker order")
 	}
+
+	// Make sure the maker order is not a FOK order.
+	if makerOrder.MustGetOrder().TimeInForce == Order_TIME_IN_FORCE_FILL_OR_KILL {
+		return errors.New("FOK order cannot be matched as a maker order")
```

Comment:
Nit: define errors in https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/types/errors.go.